k9mail/src/main/java/com/fsck/k9/activity/setup/checksettings/CheckSettingsPresenter.java

```diff
@@ -138,12 +145,25 @@ private void checkOutgoing() {
         new CheckOutgoingTask(account, new CheckSettingsSuccessCallback() {
             @Override
             public void onCheckSuccess() {
-                //We've successfully checked outgoing as well.
-                account.setDescription(account.getEmail());
                 account.save(Preferences.getPreferences(K9.app));
-                K9.setServicesEnabled(K9.app);
+                if (!editSettings) {
+                    //We've successfully checked outgoing as well.
+                    account.setDescription(account.getEmail());
+                    K9.setServicesEnabled(K9.app);
 
-                view.goToNames();
+                    view.goToNames();
+                } else {
+                    boolean isPushCapable = false;
```

Comment:
`CheckSettingsPresenter` now has some logic unrelated to checking settings. But it seems that there is no better place than it.